Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -521,39 +521,88 @@ class DropDownMenu
     var left = details.align === 'left' ? Math.max (0, LayoutUtility.GetOffset (referenceElement).left + details.offsetX) + 'px' : 'auto';
     var right = details.align === 'right' ? Math.max (0, document.documentElement.clientWidth - LayoutUtility.GetOffset (referenceElement).left - referenceElement.offsetWidth - details.offsetX) + 'px' : 'auto';
 
-    const verticalSpacing = parseInt(getComputedStyle(popUpDiv).fontSize) * 2;
+    const popUpMargin = parseInt(getComputedStyle(popUpDiv).fontSize) * 2;
 
     // Save and restore the scrollTop value to retain the scrolling after resizing
     var scrollTop = popUpDiv.scrollTop;
 
     popUpDiv.style.top = top + 'px';
     popUpDiv.style.bottom = 'auto';
-    popUpDiv.style.right = right;
-    popUpDiv.style.left = left;
     popUpDiv.style.position = 'fixed';
 
-    // move dropdown if there is not enough space to fit it on the page
-    if ((LayoutUtility.GetWidth (popUpDiv) > space_left) && (space_left < space_right))
+    // Set to values that allow the popup to expand properly - this way the result of GetWidth is actually correct
+    popUpDiv.style.left = '0';
+    popUpDiv.style.right = 'auto';
+
+    const isContextMenu = details.align === 'left';
+    const requiredPopUpWidth = LayoutUtility.GetWidth(popUpDiv);
+    const dropDownMenuButtonWidth = LayoutUtility.GetWidth(referenceElement);
+    const maxLeftAlignedPopupWidth = (space_right - popUpMargin) + dropDownMenuButtonWidth;
+    const maxRightAlignedPopupWidth = (space_left - popUpMargin) + dropDownMenuButtonWidth;
+    const isPopUpSmallerThanViewPort = requiredPopUpWidth <= (document.documentElement.clientWidth - 2 * popUpMargin);
+
+    const alignToLeftOfReferenceElement = () =>
     {
-      if (LayoutUtility.GetOffset (popUpDiv).left < 0)
-      {
-        left = Math.max (0, LayoutUtility.GetOffset (referenceElement).left) + 'px';
-        popUpDiv.style.left = left;
-        popUpDiv.style.right = 'auto';
-      }
+      popUpDiv.style.left = left;
+      popUpDiv.style.right = 'auto';
+    };
+    const alignToRightOfReferenceElement = () =>
+    {
+      popUpDiv.style.left = 'auto';
+      popUpDiv.style.right = right;
+    };
+    const alignToLeftViewPortBorder = () =>
+    {
+      popUpDiv.style.left = popUpMargin + 'px';
+      popUpDiv.style.right = 'auto';
+    };
+    const alignToRightViewPortBorder = () =>
+    {
+      popUpDiv.style.left = 'auto';
+      popUpDiv.style.right = popUpMargin + 'px';
+    };
+    const clampToViewPort = () =>
+    {
+      popUpDiv.style.left = popUpMargin + 'px';
+      popUpDiv.style.right = popUpMargin + 'px';
+    };
+
+    if (isContextMenu && (requiredPopUpWidth <= maxLeftAlignedPopupWidth))
```

Comment:
I'm afraid the context menu positoning isn't working yet:
![image](https://user-images.githubusercontent.com/6972548/164263167-04a47358-3e5c-4e5b-8d97-723614430110.png)